Fixes in cmake scripts matching the recent changes in the repo layout. (#322)
RHVoice/RHVoice: src/third-party/sonic/CMakeLists.txt

# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <https://www.gnu.org/licenses/>.
find_package(Sonic REQUIRED)
set(SONIC_DIR "${CMAKE_SOURCE_DIR}/external/libs/sonic")

if(SONIC_FOUND)

> 	message(STATUS "Sonic found, no need to recompile, using prom the distro packages. Set Sonic_FORCE_COMPILATION if you want to enforce recompilation")

Reviewer: ```suggestion
	message(STATUS "Sonic found, no need to recompile, using from distro package. Set Sonic_FORCE_COMPILATION if you want to enforce recompilation")
```

> 	message(STATUS "Sonic found, no need to recompile, using the one from the packages of your distro. Set Sonic_FORCE_COMPILATION if you want to enforce recompilation")

Reviewer: i think this message still isn't correct.
its overly verbose and doesn't read as console output.
  
```suggestion
	message(STATUS "Sonic found, using from the distribution package. Set Sonic_FORCE_COMPILATION if you want to use it from a submodule.")
```

Author: Thanks for 6he correction, this way is rewlly better. (but technically, it is from multiple packages, one for ueaders and another one for a shared library)

Reviewer: you forgot to apply my correction, didn't you?

Author: Yeah, forgot to `git add`.

	set_target_properties(sonic PROPERTIES VERSION "${LIBSONIC_VERSION}" SOVERSION "${LIBSONIC_VERSION_MAJOR}")
	set(CPACK_COMPONENT_LIBSONIC_VERSION_MAJOR "${LIBSONIC_VERSION_MAJOR}")
	set(CPACK_COMPONENT_LIBSONIC_VERSION_MINOR "${LIBSONIC_VERSION_MINOR}")
	set(CPACK_COMPONENT_LIBSONIC_VERSION_PATCH "${LIBSONIC_VERSION_PATCH}")
	set(CPACK_COMPONENT_LIBSONIC_VERSION "${LIBSONIC_VERSION}")

	target_include_directories(sonic PUBLIC "${SONIC_DIR}")
	harden(sonic)
	add_sanitizers(sonic)
	set(CPACK_DEBIAN_LIBSONIC_PACKAGE_NAME "RHVoice-libsonic")
	set(CPACK_DEBIAN_LIBSONIC_PACKAGE_REPLACES "${DEB_LIBSONIC_PACKAGES}")
	set(CPACK_DEBIAN_LIBSONIC_PACKAGE_CONFLICTS "${CPACK_DEBIAN_LIBSONIC_PACKAGE_REPLACES}")
	set(CPACK_DEBIAN_LIBSONIC_PACKAGE_BREAKS "${CPACK_DEBIAN_LIBSONIC_PACKAGE_REPLACES}")
else()
	add_library(sonic SHARED IMPORTED)
	set_property(TARGET sonic PROPERTY IMPORTED_LOCATION "${Sonic_LIBRARIES}")
	target_include_directories(sonic INTERFACE "${Sonic_INCLUDE_DIRS}")
	set(CPACK_DEBIAN_LIBSONIC_PACKAGE_NAME "${DEB_LIBSONIC_PACKAGE}")
endif()

if(WIN32 OR Sonic_FORCE_COMPILATION)
	cpack_add_component(libsonic 
		DISPLAY_NAME "libsonic"

> 		DESCRIPTION "libsonic - a library for changing rate of speech. This component may contain a modified version of libsonic, for the modifications see the RHVoice GitHub repo. The original resides here: https://github.com/waywardgeek/sonic"

Reviewer: it is submodule so cannot contain any modifications.

Author: It is not, the submodule itself is in another dir.

Reviewer: yes it isn't in third-party it is in: set(SONIC_DIR "${CMAKE_SOURCE_DIR}/external/libs/sonic")
but it cannot contain any modifications.

Author: I don't understand what you mean. `CMakeLists.txt` for sonic is in this repo and is not a part of `sonic` repo (though probably we should make a PR creating one in their repo doing proper packaging of libsonic, not the way it is curently done in Debian/Ubuntu (currently it is without pkg-config and CMake support (`FindSonic.cmake` should not be a part of this repo, instead it should be installed as a part of `libsonic-dev`)).

Reviewer: ok now I see. but I still think the description is wrong.

> we should make a PR creating one in their repo doing proper packaging of libsonic, not the way it is curently done in Debian/Ubuntu 

yep. i want to use meson for that. i have WIP branch need to finish it.

Author: >i want to use meson for that. 

You may want to take a look at https://github.com/prebuilder/prebuilder.py . Also python-based and has more functionality for packaging. Unfortunately it has 3 flaws. 1. it got broken somewhen and I haven't yet fixed it. 2. it is not mature and not very finished  (i.e. the backends for anything except Debian are not really implemented, because I only use Ubuntu) and api-stable 3. It requires some changes in third-party software, such as dpkg, and these changes have not yet been accepted, they have even not been reviewed. I don't know what to do with that except either stating "one must use an our fork of apt" or maybe just extremely inefficiently working around the missing functionality.
